src/js/hooks/list-pages/purchase-order/usePurchaseOrderFilters.jsx

```diff
@@ -25,7 +27,9 @@ const usePurchaseOrderFilters = () => {
     paymentTerms,
     currentUser,
     currentLocale,
+    translate,
   } = useSelector(state => ({
+    translate: translateWithDefaultMessage(getTranslate(state.localize)),
```

Comment:
We are using `translateWithDefaultMessage(getTranslate(state.localize))` really often (I took this function just for example). What do you think about creating a place for all of ours 'selectors'? (Maybe just as a future improvement)
After that we can do something like:
```
translate: getTranslation()
```
If there will be a need for only one selector we can skip the `state` argument and do only:
```
const translate = useSelector(getTranslation())
```
cc @kchelstowski @awalkowiak

Reply:
@alannadolny could you expand on that and show a more detailed example? Because for now I don’t see how you want to get rid of the state argument?

Reply:
*An example from one of my projects*
If I created a selector like this:
```
export const getReviewsByExerciseId = (id) => (state) =>
  state.reviews.reviews.filter((review) => review.exercise === id);
```
There is no need to pass the `state` argument, because it will be applied automatically. In this example we have to pass the `id` only.
So, it will be used in this way:
```
const reviews = useSelector(getReviewsByExerciseId(id));
```
In the code fragment which I selected for comment in this review, we have to pass the `state` argument. It can be done:
```
const supportedActivities = useSelector(getSupportedActivities());
const buyers = useSelector(getBuyers());
const currentLocation = useSelector(getCurrentLocation());
```

Reply:
I quite like the idea, but I can only see it being useful in cases like `translate` where we can replace the whole boilerplate of having to import   `getTranslate` and `translateWithDefaultMessage`

Reply:
I think that using multiple `useSelector` in a component is possible so it shouldn't be problem to do something like
```js
const translate = useSelector(getTranslateSelector);
const { locale, buyers } = useSelector((state) => ({ locale: state.session.locale, buyers: state.buyers  }));

```

Reply:
yeah, I know this is possible and we currently do it like this, but I was wondering if @alannadolny has a follow-up preposition of his idea to  get rid of passing state and doing it for multiple values.

Reply:
this is what I was afraid of